### DISCUSSION TITLE
ERROR: Cannot execute prepared SQL command when using JDBC driver

### DISCUSSION THREAD

**Discussion by RicardoFernandez-UY** [dbeaver/dbeaver#15254] [Q&A]
Hi,
I'm using a JDBC driver to connect to a data source (HPCCSystems), and after configuring the driver it connects.
I added the driver using the driver manager, as I have the .jar file. I'm working on Windows 10.

The connection validates, and also reads and shows the files/tables present in the data source, but when I ask it to retrieve the data in the tables it throws this error.

Cannot execute prepared SQL command,  SQL Query seems to be empty!
But neither the query nor the table is empty, as I execute the same query from another client (DBVisualizer) using the same driver and the same default configuration, and it returns 1M rows.

I've tried many different variations of configurations and now I really don't know what else to try to make it work in DBeaver.
I will really appreciate any help in getting this to work!
I also have many other data connections to other databases working fine in DBeaver.

Below I paste the error stack:

Thank you!

```
java.sql.SQLException: 	:  Cannot execute prepared SQL command
	SQL Query seems to be empty!
	at org.hpccsystems.jdbcdriver.HPCCPreparedStatement.executeQuery(HPCCPreparedStatement.java:104)
	at org.hpccsystems.jdbcdriver.HPCCPreparedStatement.execute(HPCCPreparedStatement.java:360)
	at org.hpccsystems.jdbcdriver.HPCCStatement.execute(HPCCStatement.java:208)
	at org.jkiss.dbeaver.model.impl.jdbc.exec.JDBCStatementImpl.execute(JDBCStatementImpl.java:330)
	at org.jkiss.dbeaver.model.impl.jdbc.exec.JDBCStatementImpl.executeStatement(JDBCStatementImpl.java:130)
	at org.jkiss.dbeaver.model.impl.jdbc.struct.JDBCTable.readData(JDBCTable.java:190)
	at org.jkiss.dbeaver.ui.controls.resultset.ResultSetJobDataRead.lambda$0(ResultSetJobDataRead.java:118)
	at org.jkiss.dbeaver.model.exec.DBExecUtils.tryExecuteRecover(DBExecUtils.java:171)
	at org.jkiss.dbeaver.ui.controls.resultset.ResultSetJobDataRead.run(ResultSetJobDataRead.java:116)
	at org.jkiss.dbeaver.ui.controls.resultset.ResultSetViewer$ResultSetDataPumpJob.run(ResultSetViewer.java:4810)
	at org.jkiss.dbeaver.model.runtime.AbstractJob.run(AbstractJob.java:105)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:63)
	

eclipse.buildId=unknown
java.version=11.0.12
java.vendor=Eclipse Foundation
BootLoader constants: OS=win32, ARCH=x86_64, WS=win32, NL=en_US
Command-line arguments:  -os win32 -ws win32 -arch x86_64
```

**Comment by LonwoLonwo** [comment]
Hello @RicardoFernandez-UY 

1. Can you tell me more about your driver?
2. As I can see in your error log - the issue appeared when you tried to open the table. Can you open your table from the [SQL Editor](https://github.com/dbeaver/dbeaver/wiki/SQL-Editor)? With the simple "Select * from table"?

**Reply by RicardoFernandez-UY** [reply]
Hi @LonwoLonwo ,
Thank you very much for your reply.

1. This is a link to the source code of the driver. I can't tell you details myself because I'm just a user, I don't really know how it works internally.   ` https://github.com/hpcc-systems/hpcc-jdbc`
2.  Yes I've tried exactly that, and it doesn't work. It shows the same error.
     Apparently anything that attempts to retrieve information from within the tables (that are actually files) fails. However it is capable of retrieving metadata like the table names, and field names and types.

Please, consider that when using the same driver in the other generic client I mentioned, it works without any problem, and retrieves the information from the tables. That is why so far I've discarded the driver itself as the problem.
I believe that as there are so many configuration options for the driver and the connection itself, I might be missing something, or perhaps one of the defaults chosen by DBeaver is causing the problem, but if this is the case I have failed to determine which one is the one. And also the configuration in DBVisualizer is a bit different so, although I've tried,  I couldn't map all the options one to one, to see if I could spot the difference.

Warm regards!

**Reply by RicardoFernandez-UY** [reply]
Hi,
Was the information I added helpful to advance in troubleshooting the issue?
Thank you.

**Reply by LonwoLonwo** [reply]
Unfortunately, I don't have a test environment. Maybe you can give me some advice about HPCC installation? On Windows.
